pprofile.py

```diff
@@ -883,14 +962,37 @@ def runpath(path, argv, filename=None, threads=True, verbose=False):
     """
     _run(threads, verbose, 'runpath', filename, path, argv)
 
-_allsep = os.sep + (os.altsep or '')
+def _qCacheGringAdapt(path):
+    '''
+    qCacheGrind (windows build) needs at least one UNIX separator
+    in path to find the file. Adapt here even if this is probably
+    more of a qCacheGrind issue...
+    '''
+
+    if os.path.sep != "/":
+        path = lambda x, cascade=path: cascade(
+            '/'.join(x.split(os.path.sep))
+        )
+
+    return path
+
+def _path_resolver(relative_path=True):
+    '''
+    Returns a function that manipulates a path string to be relative if
+    `relative_path' is true, otherwise it returns the identity function.
```

Comment:
I guess that depends on what you're looking at - and please correct me if I'm wrong :)

- For readability, I think a call to a function with a clear name is better than a block of inline code.
- For performance, if the inline code is actually substantially more performant than the function call - then the inline code makes sense.

On `_qCacheGringAdapt()` - it is only called from two functions (get_stats(), and callgrind()) , and both these functions are also only called once per run.

On `_path_resolver()` - it is only called from three functions, but again all three functions (get_stats(), callgrind(), and `main()`) are only called once per run of the utility.

Finally, on the default behavior - I think you're right, but what if we simply make that argument more communicative, and mandatory; i.e.:

 - From `def _path_resolver(relative_path=True):`
 - To `def _path_resolver(is_relative_path):`